x/observer/keeper/ballot.go

```diff
@@ -58,10 +73,41 @@ func (k Keeper) GetAllBallots(ctx sdk.Context) (voters []*types.Ballot) {
 
 // AddBallotToList adds a ballot to the list of ballots for a given height.
 func (k Keeper) AddBallotToList(ctx sdk.Context, ballot types.Ballot) {
-	list, found := k.GetBallotList(ctx, ballot.BallotCreationHeight)
+	list, found := k.GetBallotListForHeight(ctx, ballot.BallotCreationHeight)
 	if !found {
 		list = types.BallotListForHeight{Height: ballot.BallotCreationHeight, BallotsIndexList: []string{}}
 	}
 	list.BallotsIndexList = append(list.BallotsIndexList, ballot.BallotIdentifier)
 	k.SetBallotList(ctx, &list)
 }
+
+// ClearMaturedBallotsAndBallotList deletes all matured ballots and the list of ballots for a given height.
+// It also emits an event for each ballot deleted.
+func (k Keeper) ClearMaturedBallotsAndBallotList(ctx sdk.Context, maturityBlocksParam int64) {
+	maturedBallotsHeight := getMaturedBallotHeight(ctx, maturityBlocksParam)
+
+	// Fetch all the matured ballots, return if no matured ballots are found
+	// For the current implementation, this should never happen as ClearMaturedBallotsAndBallotList is only called after the Distribution of the rewards,
+	//	which means that there are matured ballots to be deleted
+	maturedBallots, found := k.GetBallotListForHeight(ctx, maturedBallotsHeight)
+	if !found {
+		return
+	}
+	// Delete all the matured ballots and emit an event for each ballot deleted
+	for _, ballotIndex := range maturedBallots.BallotsIndexList {
+		ballot, found := k.GetBallot(ctx, ballotIndex)
+		if !found {
+			continue
+		}
+		k.DeleteBallot(ctx, ballotIndex)
```

Comment:
what happens if the ballot is still pending with minority votes (e.g., 4/9)?

Reply:
I think the chances of that are quite unlikely, that is why we have maturity blocks 
However, if that does happen, a new ballot would be created when the fifth vote comes in, and the four observers who already voted would need to revote. This can be triggered by adding trackers manually.

Overall though the maturity blocks should be adjusted so that it has enough time for a ballot to finalize as we do not want to keep the ballots in the system indefinitly